storage/replica_raftstorage.go

```diff
@@ -193,8 +192,13 @@ func (r *Replica) loadAppliedIndex(eng engine.Engine) (uint64, error) {
 		return 0, err
 	}
 	if v != nil {
-		var err error
-		_, appliedIndex, err = encoding.DecodeUint64(v.GetRawBytes())
+		bytes, err := v.GetBytes()
+		if err != nil {
+			return 0, err
+		}
+		// Cannot use v.GetInt() here because we need the precision of
+		// uint64; GetInt() is lossy.
```

Comment:
Related: I think `setAppliedIndex` should use `roachpb.Value.SetInt(int64(appliedIndex))`.